From 77436089203a8a890619e3bfb6ca22634d8fc20b Mon Sep 17 00:00:00 2001
From: Varsha Maira <vmaira@habana.ai>
Date: Wed, 28 May 2025 10:45:10 +0300
Subject: [PATCH] [SW-230471] specs: add enum to define protection bit types

This patch adds an enum to represent protection levels for register
access. This helps distinguish between privilege-based protection
(blocks secure and user initiators) and secure-based protection
(blocks only user initiators). This will allow more granular access
control where needed.

Change-Id: I97c0c719a25cb72c081d07a3c97f674b8db18f0d
Signed-off-by: Varsha Maira <vmaira@habana.ai>
---

diff --git a/hl_protbits_fw_if.h b/hl_protbits_fw_if.h
index c145bb7..aee91bf 100644
--- a/hl_protbits_fw_if.h
+++ b/hl_protbits_fw_if.h
@@ -16,6 +16,19 @@
 #define HL_CNF_ALL_NON_PROT		0xFFFFFFFF
 /* GLBL_PRIV register offset from the start of the block */
 #define HL_GLBL_PRIV_REG_OFFSET		0xE80
+/* GLBL_SEC register offset from the start of the block */
+#define HL_GLBL_SEC_REG_OFFSET		0xF80
+
+/*
+ * Type of protection bit configuration.
+ * - Privilege: Blocks access from secure and user initiators.
+ * - Secure: Blocks access from user initiators only; allows secure and privileged access.
+ */
+enum prot_bit_priv_level {
+	PROT_BITS_PRIVILEGE,
+	PROT_BITS_SECURED,
+	PROT_BITS_USER,
+};
 
 /*
  * Type of the IP block with protection bit configuration.
